spark/src/main/scala/org/apache/spark/sql/delta/catalog/DeltaCatalog.scala

```diff
@@ -692,6 +705,16 @@ class DeltaCatalog extends DelegatingCatalogExtension
           })
         }
 
+        columnChanges.foreach {
+          case change: ColumnChange =>
+            val field = change.fieldNames()
+            val (oldField, _) = getColumn(field)
+            if (!allowedIdentityColumnChanges.contains(change.getClass) &&
+              ColumnWithDefaultExprUtils.isIdentityColumn(oldField)) {
+              throw DeltaErrors.identityColumnAlterColumnNotSupported()
+            }
```

Comment:
The way to do this with case matching directly is
```suggestion
            val allowsIdentityColumnChanges = change match {
              case _: UpdateColumnComment | _: UpdateColumnPosition | _: RenameColumn | _: SyncIdentity => true
              case _ => false
            }
            if (!allowsIdentityColumnChange &&
              ColumnWithDefaultExprUtils.isIdentityColumn(oldField)) {
              throw DeltaErrors.identityColumnAlterColumnNotSupported()
            }
```

or the other way to do this is at the top level (my preference)
```
 // Any ColumnChange not explicitly on the allowlist is blocked from making changes on Identity Columns 
 val disallowedColumnChangesOnIdentityColumns = columnChanges.filterNot {
   change match {
          case _: UpdateColumnComment | _: UpdateColumnPosition | _: RenameColumn | _: SyncIdentity => true
          case _: => false
   }
  }
 disallowedColumnChangesOnIdentityColumns.forEach {
  ... 
 }

```